vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state.go

```diff
@@ -197,9 +197,19 @@ func (a *AggregateContainerState) AddSample(sample *ContainerUsageSample) {
 	case ResourceMemory:
 		a.AggregateMemoryPeaks.AddSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
 	case ResourceRSS:
-		a.AggregateRSSPeaks.AddSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
+		// Special OOM handling for binary decaying histogram.
+		if sample.isOOM {
+			a.AggregateRSSPeaks.AddOomSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
+		} else {
+			a.AggregateRSSPeaks.AddSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
+		}
 	case ResourceJVMHeapCommitted:
-		a.AggregateJVMHeapCommittedPeaks.AddSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
+		// Special OOM handling for binary decaying histogram.
+		if sample.isOOM {
+			a.AggregateJVMHeapCommittedPeaks.AddOomSample(BytesFromMemoryAmount(sample.Usage), 1.0, sample.MeasureStart)
```

Comment:
iiuc we don't currently add any samples for jvm heap. What is the rationale here for only adding one for native memory and rss, but not heap?

Reply:
(discussed offline, summarizing) The `jvmheap.AddOomSample` here handles the setup to add OOM samples for JVM heap the same way as RSS, i.e. add to the next higher histogram bucket, since they both recommend the max "usage" historically. 

However, this logic wouldn't be triggered yet given that we don't know what the exact JVM OOM signal will be so nothing adds JVM OOM samples. We do have real samples for vpa-test-service - @jodzga enabled container-level for this one a while back